code/modules/projectiles/ammo_datums.dm

```diff
@@ -1358,6 +1358,40 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
 	accuracy_var_low = 3
 	accuracy_var_high = 3
 
+/datum/ammo/energy/plasma_pistol
+	name = "ionized plasma bolt"
+	icon_state = "overchargedlaser"
+	hud_state = "laser_sniper"
+	damage = 20
+	max_range = 40
+	penetration = 50
+	sundering = 5
+
+	flags_ammo_behavior = AMMO_ENERGY|AMMO_INCENDIARY|AMMO_SUNDERING|AMMO_EXPLOSIVE
+
+	bullet_color = COLOR_VIBRANT_LIME
+
+	var/heat = 12
+	var/burn = 6
+	var/f_color = "green"
+
+/datum/ammo/energy/plasma_pistol/on_hit_turf(turf/T, obj/projectile/proj)
+	if(!istype(T))
+		return
+	T.ignite(heat, burn, f_color)
```

Comment:
I don't know projectile code enough, but are you not igniting the turf you are hitting and not the turf before? 
like if it hits a wall, it will ignite the wall?